pylint/checkers/similar.py

```diff
@@ -152,7 +164,9 @@ def stripped_lines(lines, ignore_comments, ignore_docstrings, ignore_imports):
                     docstring = None
                 line = ''
         if ignore_imports:
-            if line.startswith("import ") or line.startswith("from "):
+            if lineno in line_is_import:
```

Comment:
These two lines could be rewritten as `current_line_is_import = line_is_import.get(lineno)`

Reply:
It can't:  When `lineno` is not in `line_is_import`, `current_line_is_import` needs to keep its previous true/false value.  Many lines are not present in line_is_import.  I those cases, we must continue to use the value from the previous line.

`current_line_is_import = line_is_import.get(lineno, current_line_is_import)` would work, but this doesn't read as clearly to me.

I renamed `line_is_import` to `line_begins_import` to try to clarify this.

Reply:
Oh gotcha, this makes sense then. But I definitely suggest squashing this into `line_is_import.get(lineno, current_line_is_import)` as there is no reason to do the lookup and the retrieval separately.

Reply:
Um, ok.  Done.

(If you don't mind making this a teaching moment: Why?  I thought it might be for performance, but I measured and found that using .get() is slower (p=0.002, 95% CI slower by 130 - 570 ns/line) in cpython 3.6 (it trades a LOAD_FAST, COMPARE_OP, POP_JUMP_IF_FALSE, and BINARY_SUBSCR for a LOAD_ATTR, CALL_FUNCTION, and whatever the function does).  Is it expected to be faster in pypy and people who care about performance run pylint with pypy, or is this just a python idiom I need to absorb?  Thanks!)

Reply:
Oh, to be honest I was under the impression that this would have been faster than a lookup plus an additional access, put apparently that's not the case! So instead if was a teaching moment for me. :) Nonetheless, reducing the two lines to one looks better to me.